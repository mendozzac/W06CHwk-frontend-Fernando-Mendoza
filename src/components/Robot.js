const Robot = ({ name, features }) => {
  return (
    <>
      <div>
        <img src="url" alt="robot"></img>
        <p>{name}</p>
        <p>{features}</p>
      </div>
    </>
  );
};

export default Robot;
