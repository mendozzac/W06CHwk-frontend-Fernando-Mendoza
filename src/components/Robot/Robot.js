const Robot = ({ robot }) => {
  return (
    <>
      <li>
        <div className="robot-box">
          <img src={robot.image} width="200px" height="200px" alt="robot"></img>
          <p>{robot.name}</p>
          <p>Velocidad: {robot.features.speed}</p>
          <p>Resistencia: {robot.features.stamina}</p>
        </div>
      </li>
    </>
  );
};

export default Robot;
