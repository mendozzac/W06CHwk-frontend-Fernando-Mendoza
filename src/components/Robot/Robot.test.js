import { render, screen } from "@testing-library/react";
import ReactTestRenderer from "react-test-renderer";
import Robot from "./Robot";

describe("Given a Robot component", () => {
  describe("When it receives a robot", () => {
    test("Then it should render the robot", () => {
      const robot = {
        name: "Fredy",
        image: "plastic",
        features: {
          speed: 5,
          stamin: 6,
        },
      };

      render(<Robot robot={robot} />);
      const newRobot = screen.getByText("Fredy");

      expect(newRobot).toBeInTheDocument();
    });
  });
  describe("When it receives an object robot", () => {
    test("Then it should render the robot", () => {
      const robot = {
        name: "Fredy",
        image: "plastic",
        features: {
          speed: 5,
          stamin: 6,
        },
      };

      const newRobot = ReactTestRenderer.create(<Robot robot={robot} />);
      expect(newRobot.toJSON()).toMatchSnapshot();
    });
  });
});
