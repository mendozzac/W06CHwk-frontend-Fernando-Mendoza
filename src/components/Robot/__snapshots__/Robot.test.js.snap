// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Robot component When it receives an object robot Then it should render the robot 1`] = `
<li>
  <div
    className="robot-box"
  >
    <img
      alt="robot"
      height="200px"
      src="plastic"
      width="200px"
    />
    <p>
      Fredy
    </p>
    <p>
      Velocidad: 
      5
    </p>
    <p>
      Resistencia: 
    </p>
  </div>
</li>
`;
